### ISSUE TITLE
Alert Manager fails to query Azure Monitor for an Event Hub with EntityName filter

### ISSUE THREAD

**Issue by sflandergan** [grafana/grafana#41243]
**What happened**:

I tried to create an `alert` for the `Incoming Messages` metric of an Azure `Event Hub Namespace` with `EntityName` filter.
While the `query` executes successfully and the graph shows the metric for the filtered `Event Hub` the `Alert Manager` fails to query the metric for evaluation with following error:

```json
{
  "firing": true,
  "state": "pending",
  "conditionEvals": " = true",
  "timeMs": "25.500ms",
  "error": "request handler response error {request failed, status: 400 Received invalid query parameter: [$filter, EntityName+eq+%27asset-messages%27],[aggregation, Total],[api-version, 2018-01-01],[interval, PT1M],[metricnames, IncomingMessages],[metricnamespace, microsoft.eventhub%2Fnamespaces],[timespan, 2021-11-03T09%3A41%3A15Z%2F2021-11-03T09%3A46%3A15Z],[top, ], TraceId: {103c6b9a-5dfe-443b-a321-7dedcd76766f} for request: subscriptions/<hidden>/resourceGroups/<hidden>/providers/Microsoft.eventhub/namespaces/<hidden>/providers/mic  A <nil> [] [] 0xc002025da0}",
  "logs": [
    {
      "message": "Condition[0]: Query",
      "data": {
        "from": 1635932475178,
        "queries": [
          {
            "refId": "A",
            "model": {
              "azureMonitor": {
                "aggregation": "Total",
                "dimensionFilters": [
                  {
                    "dimension": "EntityName",
                    "filter": "asset-messages",
                    "operator": "eq"
                  }
                ],
                "metricDefinition": "Microsoft.eventhub/namespaces",
                "metricName": "IncomingMessages",
                "metricNamespace": "microsoft.eventhub/namespaces",
                "resourceGroup": "<hidden>",
                "resourceName": "<hidden>",
                "timeGrain": "auto"
              },
              "queryType": "Azure Monitor",
              "refId": "A",
              "subscription": "<hidden>"
            },
            "datasource": {
              "id": 3,
              "name": "azure-monitor"
            },
            "maxDataPoints": 1500,
            "intervalMs": 200
          }
        ],
        "to": 1635932775178
      }
    }
  ]
}
```

**What you expected to happen**:

Alert rule should execute successfully

**How to reproduce it (as minimally and precisely as possible)**:

- Create an `empty panel `
- Pick an `Azure Monitor` data source
- Select for Query 'A':
  - `subscription = <your azure subscription>`
  - `resource group = <your resource group having an event hub namespace deployed>`
  - `resource type = Event Hubs Namespace`
  - `resource name = <your event hub namespace>`
  - `metric namespace = microsoft.eventhub-namespaces`
  - `metric = Incoming Messages`
  - Click `Add new dimension`
    -  `EntityName == <name of an event hub>`
- Switch to `Alert` tab
- Click `Create Alert`
- Add `IS  ABOVE = 0`
- Click `Test Rule`

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.2.1
- Data source type & version: Azure Monitor 0.3.1
- OS Grafana is installed on: Docker Container with image `grafana/grafana:8.2.1` running on AKS with k8s v.1.19.3
- User OS & Browser: Ubuntu 20.04, Google Chrome 94.0.4606.81
- Grafana plugins: only core plugins

**Comment by mattmichal** [comment]
Seeing a similar issue on Grafana 8.2.2 with Azure Monitor 0.3.1.

```
request handler response error {request failed, status: 400 Received invalid query parameter: [$filter, availabilityResult%2Fname+eq+%27getstatuscode-com%27+and+availabilityResult%2Flocation+eq+%27%2A%27],[aggregation, Average],[api-version, 2018-01-01],[interval, PT1M],[metricnames, availabilityResults%2FavailabilityPercentage],[metricnamespace, microsoft.insights%2Fcomponents],[timespan, 2021-11-11T17%3A15%3A32Z%2F2021-11-11T17%3A20%3A32Z],[top, ], TraceId: {2325fd69-f365-4e22-b0b8-f963ca501143} for request: subscriptions/<redacted>/resou A <nil> [] [] 0xc0022f69f0}
```

**Comment by sflandergan** [comment]
> Seeing a similar issue on Grafana 8.2.2 with Azure Monitor 0.3.1.
> 
> ```
> request handler response error {request failed, status: 400 Received invalid query parameter: [$filter, availabilityResult%2Fname+eq+%27getstatuscode-com%27+and+availabilityResult%2Flocation+eq+%27%2A%27],[aggregation, Average],[api-version, 2018-01-01],[interval, PT1M],[metricnames, availabilityResults%2FavailabilityPercentage],[metricnamespace, microsoft.insights%2Fcomponents],[timespan, 2021-11-11T17%3A15%3A32Z%2F2021-11-11T17%3A20%3A32Z],[top, ], TraceId: {2325fd69-f365-4e22-b0b8-f963ca501143} for request: subscriptions/<redacted>/resou A <nil> [] [] 0xc0022f69f0}
> ```
> 
> #41001 could be related?

Seems this is not the same as #41001. 
I just upgraded to Grafana 8.2.5 to check if this one is fixed too.
Unfortunately I can still reproduce the bug in 8.2.5

**Comment by Nighthawk22** [comment]
Same in Grafana 8.3.0 sadly. Is there any workaround for alerting? :/

**Comment by eugenecherepanov** [comment]
Hi all. Same issue  Grafana v8.3.3 (30bb7a93ca)

> Same in Grafana 8.3.0 sadly. Is there any workaround for alerting? :/
I want to try export metric to prometheus with azure-metrics-exporter. And after set up alert with prometheus datasource

**Comment by yuri-tceretian** [comment]
I was able to reproduce it. The problem seems to be in dimension filtering. Once I removed the dimension filter evaluation stopped failing.

**Comment by coops1106** [comment]
Just stumbled across this same issue, which is a shame because the company I work for are currently undergoing a queue migration to azure service bus.  Can I feed back to my higher ups, that this issue is being worked on?
Thanks

**Comment by sflandergan** [comment]
Thanks @coops1106 !

**Comment by vkamlesh** [comment]
> @sflandergan hey buddy, just in case the above is still a hindrance for you. We contacted grafana's support department, who gave us a work around. Turns out if you add Top and a value to (we added 1, they suggested 5 or 10) to the metric you are alerting on, it works. HTHs

Cool trick @coops1106 but It will not work if you are using Template Variables in the dashboard.

**Comment by andresmgot** [comment]
Hi, this was fixed in https://github.com/grafana/grafana/pull/47010 and released in 8.4.5. Please let us know if you still find the issue using that version!

**Comment by coops1106** [comment]
> Hi, this was fixed in #47010 and released in 8.4.5. Please let us know if you still find the issue using that version!

Hey, sorry it's taken me a while to check our version we're using v8.5.0 (6134e3cf3)

**Comment by andresmgot** [comment]
hi @coops1106, the fix should be present in v8.5.0 so let us know if that's not the case for you.
